app/javascript/controllers/tab_error_controller.js

```diff
@@ -11,6 +11,9 @@ export default class extends Controller {
       const paneEl = this.paneTargets[index]
       if (paneEl.querySelector('.is-invalid')) {
         tabEl.classList.add('is-invalid')
+        tabEl.setAttribute('aria-invalid', 'true')
+        tabEl.setAttribute('aria-label', 'Error: required fields missing')
+        tabEl.setAttribute('role', 'img')
       }
```

Comment:
```suggestion
      if (paneEl.querySelector('.is-invalid')) {
        tabEl.classList.add('is-invalid')
        tabEl.setAttribute('aria-description', 'Error: required fields missing')
      } else {
        tabEl.classList.remove('is-invalid')
        tabEl.removeAttribute('aria-description')
      }
```

I don't think we want to set `aria-invalid` or a `role` here. The `aria-invalid` attribute is a standin for HTML input validation behavior; it doesn't really have an effect if set on things that aren't inputs because they can't be "invalid". Setting `role=img` also doesn't make sense because this element is a tab, not an image.

Instead, we can just add an additional description signaling that this tab contains errored fields, which is what the red exclamation mark means.

Reply:
@thatbudakguy do you think that is sufficient based on SODA's "how to fix" suggestions? If so, awesome!

Reply:
Yes – their suggested solutions are all for making the exclamation have an alt, but this just accomplishes the same thing by adding that text to the description of the tab itself. We can't add an alt to the exclamation because it's a background image added via CSS pseudoselector.

Reply:
Ok, cool, that's very confusing though as their suggestion includes the `aria-invalid` and `aria-required` values... This stuff is just clear as mud.